### PR TITLE
[webview_flutter] Support emulator

### PR DESCRIPTION
**Pull request by bwikbs**
* Use the sw backend for emulator

Signed-off-by: MuHong Byun <mh.byun@samsung.com>

### CODE REVIEW

File: packages/webview_flutter/tizen/src/webview.cc [hunk 1]
```diff
@@ -137,6 +139,23 @@ bool GetValueFromEncodableMap(const flutter::EncodableValue& arguments,
   return false;
 }
 
+bool needsSWBackend(void) {
+  bool result = false;
+  char* value;
+  int ret;
```
Comment (bbrto21) [inline packages/webview_flutter/tizen/src/webview.cc:145]:
```suggestion
  char* value = nullptr;
  int ret = 0;
```

File: packages/webview_flutter/tizen/src/webview.cc [hunk 2]
```diff
@@ -137,6 +139,23 @@ bool GetValueFromEncodableMap(const flutter::EncodableValue& arguments,
   return false;
 }
 
+bool needsSWBackend(void) {
```
Comment (bbrto21) [inline packages/webview_flutter/tizen/src/webview.cc:142]: recommend:
```suggestion
bool needsSwBackend(void) {
```

File: packages/webview_flutter/tizen/src/webview.cc [hunk 3]
```diff
@@ -916,6 +936,47 @@ void WebView::HandleMethodCall(
...
+    }
+    result->Error("InvalidArguments", "Please set 'key' properly");
+  } else if (method_name.compare("loadHtmlString") == 0) {
+    std::string html;
+    std::string baseUrl;
+    if (!GetValueFromEncodableMap(arguments, "html", &html)) {
+      result->Error("InvalidArguments", "Please set 'html' properly");
+      return;
+    }
+    if (GetValueFromEncodableMap(arguments, "baseUrl", &baseUrl)) {
```
Comment (bbrto21) [inline packages/webview_flutter/tizen/src/webview.cc:960]: `baseUrl` is not used.

File: packages/webview_flutter/tizen/src/webview.cc [hunk 4]
```diff
@@ -137,6 +139,23 @@ bool GetValueFromEncodableMap(const flutter::EncodableValue& arguments,
...
+  bool result = false;
+  char* value;
+  int ret;
+  ret = system_info_get_platform_string("http://tizen.org/system/model_name",
+                                        &value);
+  if (ret == SYSTEM_INFO_ERROR_NONE) {
+    if (strcmp(value, "Emulator") == 0) {
+      result = true;
+    }
+  }
```
Comment (seungsoo47) [inline packages/webview_flutter/tizen/src/webview.cc:151]:
```suggestion
  if ((SYSTEM_INFO_ERROR_NONE == ret) && (0 == strcmp(value, "Emulator")) {
      result = true;
  }
```
How about reducing the number of nested if statements? And how about putting the value in front of the variable in the conditional statement?

File: packages/webview_flutter/tizen/src/webview.cc (hunk 3 above)
Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc:960]: You're right. lwe is currently unable to process it, so it only shows the log.

File: packages/webview_flutter/tizen/src/webview.cc (hunk 2 above)
Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc]: Good point!

File: packages/webview_flutter/tizen/src/webview.cc (hunk 4 above)
Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc]: Thanks! That looks good! 👍

**Review by seungsoo47 — APPROVED** [review]

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**